src/Build/Instance/RunningObjectTable.cs

```diff
@@ -52,7 +49,15 @@ public object GetObject(string itemName)
 
         private IMoniker CreateMoniker(string itemName)
         {
-            Ole32.CreateItemMoniker("!", itemName, out IMoniker mk);
+            IMoniker mk = null;
+
+            // To avoid deadlock, create Moniker in a MTA thread. And the
+            // object will be MTA
+            Thread thread = new Thread(() => Ole32.CreateItemMoniker("!", itemName, out mk));
```

Comment:
Again, creating a thread is relatively expensive. Why not use `Task.Run` to leverage the thread pool?
Also, check the current thread's apartment and do it yourself directly if it's already an MTA.